BenieSereine/wepad-backend
# #176423824 user registraction

#### What does this PR do?

- Creating new user

#### Description of Task to be completed?
- A user is able to sign up as a new user
 #### How should this be manually tested?
- open your terminal
- git clone this repo by running `git clone https://github.com/BenieSereine/wepad-backend.git`
- navigate to this branch by running `git checkout  ft-user-registration-176423824`
- run `git pull ` to make sure that you have updated code from this branch
- make a .env file and add JWTKEY , SENDGRID API KEY and POSTGRES DB Url
- run `npm run migrate` to create database migrations
- run `npm run devStart`  to start the server
- open your postman or an other software to test the api endpoint using POST `localhost:3000/api/auth/signup`
- use a form like this in the body `{
    "first_name":"Benie Sereine",
    "last_name":"Mugwaneza",
    "password":"HeyH@123",
    "email":"Benie8823@gmail.com",
    "gender":"female",
    "birth_date":"2020-12-01",
    "location":"Rwanda"
    }`
#### Any background context you want to provide?
N/A
#### Screenshots (if appropriate)
![Screenshot from 2021-01-11 14-01-05](https://user-images.githubusercontent.com/73790729/104180175-835eb300-5415-11eb-91f3-97f638a39d6c.png)
